src/tools/compiletest/src/runtest.rs

```diff
@@ -295,11 +295,9 @@ impl<'test> TestCx<'test> {
     }
 
     fn check_correct_failure_status(&self, proc_res: &ProcRes) {
-        // The value the rust runtime returns on failure
-        const RUST_ERR: i32 = 101;
-        if proc_res.status.code() != Some(RUST_ERR) {
```

Comment:
Hmm, this looks suspicious. Can you instead make the `Termination` example return 101? Or else -- maybe better -- add a header command to indicate what the status should be ?

I feel like I'd like to keep testing that most of the run-fail examples return 101.

Reply:
ok

Reply:
You can ping me on IRC or gitter. There's no docs really but creating a header command is pretty straightforward. You would add a field to this struct:

https://github.com/rust-lang/rust/blob/3bcda48a30b21e46b81a7989deb30a3ba85fb918/src/tools/compiletest/src/header.rs#L181-L182

and modify this function to initialize it:

https://github.com/rust-lang/rust/blob/3bcda48a30b21e46b81a7989deb30a3ba85fb918/src/tools/compiletest/src/header.rs#L290-L293

then in this code:

https://github.com/rust-lang/rust/blob/3bcda48a30b21e46b81a7989deb30a3ba85fb918/src/tools/compiletest/src/runtest.rs#L297

you can just read `self.props.new_field`.

Reply:
If you want to be a good citizen, you can consider contributing back to rustc-guide when you're done with some tips =)

Reply:
haha--deal!

Reply:
`failure_status` header command implemented.

Reply:
"what I learned" contributed back to rustc-guide https://github.com/rust-lang-nursery/rustc-guide/pull/53  :)